### PR TITLE
Gameplay stub

### PR DESCRIPTION
**Pull request by HussainGynai**
## Description
Gameplay stub which stubs out basic122. This is a bit of mess since the actions and skills haven't been merged, but they'll need to be change to match what's here.

### Test Case 1
1. Run test gameplay stub or create your own test

Expected result: should see 5 ticks go through the process of basic122 (2 markers and one striker which captures and shoots)

### CODE REVIEW

File: rj_gameplay/rj_gameplay/gameplay_node.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    A node which subscribes to the world_state,  game state, robot status, and field topics and converts the messages to python types.
+    """
+
+    def __init__(self, play_selector: situation.IPlaySelector, world_state: Optional[rc.WorldState] = None) -> None:
+        rclpy.init()
+        super().__init__('minimal_subscriber')
+        self.world_state_sub = self.create_subscription(msg.WorldState, '/vision_filter/world_state', self.create_partial_world_state, 10)
+        self.field_dimenstions = self.create_subscription(msg.FieldDimensions, '/config/field_dimensions', self.create_field, 10)
+        self.game_info = self.create_subscription(msg.GameState, '/referee/game_state', self.create_game_info, 10)
+        for i in range(16):
```
Comment (kylestach) [inline rj_gameplay/rj_gameplay/gameplay_node.py]: Make a constant somewhere for `NUM_ROBOTS`.

File: rj_gameplay/stp/skill/__init__.py [hunk 2]
```diff
@@ -11,19 +11,9 @@ class ISkill(ABC):
     """ Interface for Skills. """
 
     @abstractmethod
-    def create_request(self, **kwargs) -> role.RoleRequest:
```
Comment (kylestach) [inline rj_gameplay/stp/skill/__init__.py:14]: Why are we removing `create_request`?

File: rj_gameplay/stp/skill/sequence.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import py_trees
+import random
+import stp.action as action
+from typing import Optional, Any
+import stp.rc as rc
+
+class RJSequence(py_trees.composites.Sequence):
```
Comment (kylestach) [inline rj_gameplay/stp/skill/sequence.py:7]: Why is this distinct from the regular `py_trees.composites.Sequence` class?

File: rj_gameplay/rj_gameplay/tactic/stub_striker.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
+"""Contains the stub for the striker tactic. """
```
Comment (kylestach) [inline rj_gameplay/rj_gameplay/tactic/stub_striker.py:1]: Could you add TODOs in these stubs for the parts we need to fill out?

File: rj_gameplay/setup.py [hunk 5]
```diff
@@ -6,12 +6,13 @@
...
     zip_safe=True,
     maintainer="oswinso",
     maintainer_email="oswinso@gmail.com",
     description="Rewrite of the gameplay library.",
     entry_points={
-        "console_scripts": [],
+        "console_scripts": ["gameplay_node = rj_gameplay.gameplay_node:main"],
     },
-)
+)
```
Comment (kylestach) [inline rj_gameplay/setup.py]: Trailing newline lmao

File: rj_gameplay/stp/action/__init__.py [hunk 6]
```diff
@@ -74,3 +80,7 @@ def create(self, action: Type[ActionT]) -> ActionT:
             )
 
         return self._registry[action]
+
+class Ctx:
+    """Context for actions, dummy for now"""
+    ...
```
Comment (kylestach) [inline rj_gameplay/stp/action/__init__.py]: Nit: newline

File: rj_gameplay/stp/skill/cost_behavior.py [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+from rj_gameplay.action import move as Move
+import stp.rc as rc
+from typing import Optional, Any, Callable
+import numpy as np
+
+
+
+class CostBehavior(py_trees.behaviour.Behaviour):
+    """
+    A behavior for skills which use the smove action based on a heuristic function, for skills like seek or mark
```
Comment (kylestach) [inline rj_gameplay/stp/skill/cost_behavior.py]: Smove

**Review by kylestach — COMMENTED** [review]
First pass of review, a few questions/comments

File: rj_gameplay/stp/skill/__init__.py (hunk 2 above)
Comment (HussainGynai) [reply rj_gameplay/stp/skill/__init__.py:14]: Tactics should be creating the requests for skills not skills themselves.

File: rj_gameplay/stp/skill/sequence.py (hunk 3 above)
Comment (HussainGynai) [reply rj_gameplay/stp/skill/sequence.py:7]: It needs to hold a robot and context that can be passed down to its children.

File: rj_gameplay/stp/rc.py [hunk 8]
```diff
@@ -15,7 +15,7 @@ class Robot:
     to enforce that instances of this class should not be mutated."""
 
     __slots__ = [
-        "__id", "__is_ours", "__pose", "__twist", "__ball_sense_triggered",
```
Comment (liquidmetal9015) [inline rj_gameplay/stp/rc.py:18]: I think I commented on this in a previous PR as well, why is this being removed again? How will ball-sense be used in the determination of possession, settling, etc if it's not included in worldstate?

**Review by liquidmetal9015 — APPROVED** [review]
Looks good to me.

File: rj_gameplay/stp/action/__init__.py [hunk 9]
```diff
@@ -83,4 +83,5 @@ def create(self, action: Type[ActionT]) -> ActionT:
 
 class Ctx:
     """Context for actions, dummy for now"""
-    ...
\ No newline at end of file
+    ...
+
```
Comment (kylestach) [inline rj_gameplay/stp/action/__init__.py]: Not blocking, but nit: in a bunch of places you added a new line with a bunch of extra indentation, not a trailing newline. As a result, all of these files still have the "no trailing newline" marker _and_ extra crap at the end.

**Review by kylestach — APPROVED** [review]
Sure

**Pull request merged**